NeoML/src/Dnn/Layers/FullyConnectedLayer.cpp

```diff
@@ -22,194 +22,260 @@ limitations under the License.
 namespace NeoML {
 
 CFullyConnectedLayer::CFullyConnectedLayer( IMathEngine& mathEngine, const char* name ) :
-	CBaseLayer( mathEngine, name == nullptr ? "CCnnFullyConnectedLayer" : name, true ),
-	numberOfElements(0),
-	isZeroFreeTerm(false)
+	CBaseLayer( mathEngine, name == nullptr ? "CCnnFullyConnectedLayer" : name, /*isLearnable*/true ),
+	numberOfElements( 0 ),
+	isZeroFreeTerm( false )
 {
 	paramBlobs.SetSize(2);
 }
 
-CFullyConnectedLayer::~CFullyConnectedLayer()
+CFullyConnectedLayer::~CFullyConnectedLayer() = default;
+
+const CSmallMatricesMultiplyDesc* CFullyConnectedLayer::initSmallMatricesMulDescs( TSMMD type, int inputNumber,
+	int firstHeight, int firstWidth, int secondWidth, int resultWidth )
+{
+	NeoPresume( inputBlobs[0] != nullptr || inputDiffBlobs[0] != nullptr );
+	NeoPresume( outputBlobs[0] != nullptr || outputDiffBlobs[0] != nullptr );
+
+	CSmallMatricesMultiplyDesc* desc = smallMatricesMulDescs[type][inputNumber];
+	if( desc == nullptr ) {
+		desc = MathEngine().InitSmallMatricesMultiplyDesc(
```

Comment:
Wait a second... _How_ newly created desc will appear in `smallMatrixMulDescs` array?

Reply:
Thank you! My fault, fixed

Reply:
New requirement: test it for memory leaks. Right now every desc in `CFullyConnectedLayer` (which is used everywhere) is leaking...